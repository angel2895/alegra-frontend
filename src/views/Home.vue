<template>
  <search-section></search-section>
  <images-section></images-section>
</template>

<script lang="ts">
import { Options, Vue } from "vue-class-component";

import SearchSection from "@/components/home/SearchSection.vue";
import ImagesSection from "@/components/home/ImagesSection.vue";

@Options({
  components: { SearchSection, ImagesSection },
})
export default class Home extends Vue {}
</script>

<style>
.search--background img {
  width: 100%;
  filter: brightness(0.6);
}
.search--cover {
  position: absolute;
  width: 100%;
  height: auto;

  background-color: rgba(0, 0, 0, 0.5);
  height: 200px;
  top: 0px;
}
</style>
