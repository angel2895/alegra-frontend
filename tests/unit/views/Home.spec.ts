import { shallowMount } from "@vue/test-utils";
import Home from "@/views/Home.vue";

describe("Home.vue", () => {
  it("Match with snapshot", () => {
    const wrapper = shallowMount(Home);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("SearchSection and ImagesSection are mounted in Home", () => {
    const wrapper = shallowMount(Home);
    expect(wrapper.find("search-section-stub").exists()).toBeTruthy();
    expect(wrapper.find("images-section-stub").exists()).toBeTruthy();
  });
});
