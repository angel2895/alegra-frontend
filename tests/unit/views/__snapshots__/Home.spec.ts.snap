// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue Match with snapshot 1`] = `
<search-section-stub></search-section-stub>
<images-section-stub></images-section-stub>
`;
